File: py_pub_sub_pkg/py_pub_sub_pkg/twist_subscriber.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from geometry_msgs.msg import TwistStamped


class TwistSubscriber(Node):
    def __init__(self):
        super().__init__('twist_subscriber')
        self.subscription = self.create_subscription(
            TwistStamped,
            'cmd_vel',
            self.twist_callback,
            10)
        self.subscription  # prevent unused variable warning

    def twist_callback(self, msg):
        self.get_logger().info(f'Received twist - linear.x: {msg.twist.linear.x}, angular.z: {msg.twist.angular.z}')


def main(args=None):
    rclpy.init(args=args)
    node = TwistSubscriber()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: cpp_pub_sub_pkg/src/twist_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
#include <geometry_msgs/msg/twist_stamped.hpp>

class TwistSubscriber : public rclcpp::Node
{
public:
    TwistSubscriber() : Node("twist_subscriber")
    {
        subscription_ = this->create_subscription<geometry_msgs::msg::TwistStamped>(
            "cmd_vel", 10,
            std::bind(&TwistSubscriber::twist_callback, this, std::placeholders::_1));
    }

private:
    void twist_callback(const geometry_msgs::msg::TwistStamped::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(),
                    "Received twist - linear.x: %f, angular.z: %f",
                    msg->twist.linear.x, msg->twist.angular.z);
    }
    rclcpp::Subscription<geometry_msgs::msg::TwistStamped>::SharedPtr subscription_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<TwistSubscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}